### DISCUSSION TITLE
Automatic entries for nav in mkdocs.yml

### DISCUSSION THREAD

**Discussion by DorKeinath** [squidfunk/mkdocs-material#6319] [Q&A] [answered]
Is it possible not to have to write the md-site-urls into `mkdocs.yml` and instead of this all pages in my folder-structure are displayed automatically?

For example is it possible to write something like the following?

```
nav:
    - Folder One:
        - Intro: folder1/index.md
        - Content of Folder One:
            - {list all md-pages from folder1 here}
```

**Comment by alexvoss** [comment]
The default behaviour if you do not have a `nav:` section is to list the pages as they appear in the directory. However, once you specify more than an empty `nav:`section in `mkdocs.yml` you need to specify the full contents. You could write a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) that attaches itself to the [on_nav event](https://www.mkdocs.org/dev-guide/plugins/#on_nav) and modifies the navigation to suit your needs.

**Reply by alexvoss** [reply, marked as answer]
Or, as I am just learning myself by looking at another post, it seems that the [mkdocs-awesome-pages-plugin](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/) might help?

**Comment by kamilkrzyskow** [comment]
You could use the awesome-pages plugin
https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin

You'd move the nav creation to another file, and then only specify the directory name to list all pages inside it.
